sopel/modules/url.py

```diff
@@ -153,6 +153,96 @@ def shutdown(bot):
             pass
 
 
+@plugin.command('urlban', 'urlpban')
+@plugin.example('.urlpban example\\.com/\\w+', user_help=True)
+@plugin.example('.urlban example.com/path', user_help=True)
+@plugin.output_prefix('[url] ')
+def url_ban(bot, trigger):
+    """Ban a URL from auto title.
+
+    Use ``urlpban`` to ban a pattern instead of a URL.
+    """
+    url = trigger.group(2)
+
+    if not url:
+        bot.reply('This command requires an URL to ban.')
+        return
+
+    if trigger.group(1) == 'urlpban':
+        # validate regex pattern
+        try:
+            re.compile(url)
+        except re.error as err:
+            bot.reply('Invalid regex pattern: %s' % err)
+            return
+    else:
+        # escape the URL to ensure a valid pattern
+        url = re.escape(url)
+
+    patterns = bot.settings.url.exclude or []
+
+    if url in patterns:
+        bot.reply('This URL is already banned from auto title.')
+        return
+
+    # update settings
+    patterns.append(url)
+    bot.settings.url.exclude = patterns
+    bot.settings.save()
+    LOGGER.info('%s banned the URL pattern "%s"', trigger.nick, url)
+
+    # re-compile
+    bot.memory['url_exclude'] = [re.compile(s) for s in patterns]
+
+    # tell the user
+    bot.reply('This URL is now banned from auto title.')
+
+
+@plugin.command('urlunban', 'urlpunban')
+@plugin.example('.urlpunban example\\.com/\\w+', user_help=True)
+@plugin.example('.urlunban example.com/path', user_help=True)
+@plugin.output_prefix('[url] ')
+def url_unban(bot, trigger):
+    """Unban a URL from auto title.
+
+    Use ``urlpunban`` to unban a pattern instead of a URL.
+    """
+    url = trigger.group(2)
+
+    if not url:
+        bot.reply('This command requires an URL to ban.')
+        return
+
+    if trigger.group(1) == 'urlpunban':
+        # validate regex pattern
+        try:
+            re.compile(url)
+        except re.error as err:
+            bot.reply('Invalid regex pattern: %s' % err)
+            return
+    else:
+        # escape the URL to ensure a valid pattern
+        url = re.escape(url)
+
+    patterns = bot.settings.url.exclude or []
```

Comment:
See comments/suggestions in `url_ban()` above.

```suggestion
    patterns = bot.settings.url.exclude
```